tokio/src/sync/watch.rs

```diff
@@ -311,6 +345,23 @@ impl<T> Receiver<T> {
     /// could cause the send half to block. It is recommended to keep the borrow
     /// as short lived as possible.
     ///
+    /// The priority policy of the lock is dependent on the underlying lock
+    /// implementation, and this type does not guarantee that any particular policy
+    /// will be used. In particular, a producer which is waiting to acquire the lock
+    /// in `send` might or might not block concurrent calls to `borrow`, e.g.:
+    ///
+    /// <details><summary>Potential deadlock example</summary>
+    ///
+    /// ```text
+    /// // Task 1 (on thread A)    |  // Task 2 (on thread B)
+    /// let _ref1 = rx.borrow();   |
+    ///                            |  // will block
+    ///                            |  let _ = tx.send(());
+    /// // may deadlock            |
+    /// let _ref2 = rx.borrow();   |
+    /// ```
+    /// </details>
+    ///
```

Comment:
The `borrow_and_update` method takes `&self` so deadlocking on it would require calling it on two separate receivers. Still, I'm ok with keeping the comment here as well.

Reply:
Good call! I'm happy to remove this comment here as well as it doesn't make as much sense.

Reply:
I updated the comment to have two separate receivers and call `borow_and_update` in the `borrow_and_update` docs